syntax = "proto3";

package pqcwrappb;


message Secret {
  string name = 1;
  int32 version = 2;
  KeyType type = 3;
  enum KeyType {
    ml_kem_512 = 0;
    ml_kem_768 = 1;
    ml_kem_1024 = 2;    
  }
  bytes kemCipherText = 4;
  bytes publicKey = 5;
}
